File: 0845-Longest Mountain in Array/0845-Longest Mountain in Array.cpp
class Solution {
public:
    int longestMountain(vector<int>& A) {
        int left = 0, maxLen = 0;
        while (left < A.size()) {
            int mid = left;
            while (mid < A.size() - 1 && A[mid + 1] > A[mid]) {
                ++mid;
            }
            if (mid == left) {
                ++left;
                continue;
            }
            
            int right = mid;
            while (right < A.size() - 1 && A[right + 1] < A[right]) {
                ++right;
            }
            if (right > mid) {
                maxLen = max(maxLen, right - left + 1);
            }
            left = right;
        }
        return maxLen;
    }
};

File: 0801-0900/0845-Longest Mountain in Array/0845-Longest Mountain in Array.py
class Solution:
    def longestMountain(self, A: List[int]) -> int:
        left = maxLen = 0
        while left < len(A):
            mid = left
            while mid < len(A) - 1 and A[mid + 1] > A[mid]:
                mid += 1
            if mid == left:
                left += 1
                continue

            right = mid
            while right < len(A) - 1 and A[right + 1] < A[right]:
                right += 1

            if right > mid:
                maxLen = max(maxLen, right - left + 1)
            left = right
        return maxLen
